python/plugins/processing/algs/otb/OtbAlgorithm.py

```diff
@@ -207,31 +211,41 @@ def processAlgorithm(self, parameters, context, feedback):
             param = self.parameterDefinition(k)
             if param.isDestination():
                 continue
-
-            if isinstance(param, (QgsProcessingParameterRasterLayer, QgsProcessingParameterVectorLayer)):
-                if isinstance(v, QgsMapLayer):
-                    value = '"{}"'.format(v.source())
-                else:
-                    value = '"{}"'.format(v)
-            elif isinstance(param, QgsProcessingParameterMultipleLayers):
-                value = ''
-                for item in v:
-                    value += '"{}" '.format(item)
-            elif isinstance(param, QgsProcessingParameterCrs):
-                crs = self.parameterAsCrs(parameters, param.name(), context)
-                value = crs.authid().split('EPSG:')[1]
-
-            elif isinstance(param, QgsProcessingParameterEnum):
+            if isinstance(param, QgsProcessingParameterEnum):
                 value = self.parameterAsEnum(parameters, param.name(), context)
+            elif isinstance(param, QgsProcessingParameterBoolean):
+                value = self.parameterAsBool(parameters, param.name(), context)
+            elif isinstance(param, QgsProcessingParameterCrs):
+                crsValue = self.parameterAsCrs(parameters, param.name(), context)
+                value = crsValue.authid().split('EPSG:')[1]
```

Comment:
authid won't always start with "EPSG:". You may also get "USER:10001", etc. Does OTB have any way to handle user defined projections?